music_assistant/providers/squeezelite/player.py

```diff
@@ -276,35 +286,42 @@ async def play_media(self, media: PlayerMedia) -> None:
             )
         elif media.media_type == MediaType.PLUGIN_SOURCE:
             # special case: plugin source stream
+            if not media.custom_data:
+                raise InvalidDataError("Missing custom_data for plugin source")
             audio_source = self.mass.streams.get_plugin_source_stream(
                 plugin_source_id=media.custom_data["source_id"],
                 output_format=master_audio_format,
                 # need to pass player_id from the PlayerMedia object
                 # because this could have been a group
                 player_id=media.custom_data["player_id"],
             )
-        elif media.source_id.startswith("ugp_"):
+        elif media.source_id and media.source_id.startswith("ugp_"):
             # special case: UGP stream
-            ugp_player: UniversalGroupPlayer = self.mass.players.get(media.source_id)
+            ugp_player = cast("UniversalGroupPlayer", self.mass.players.get(media.source_id))
+            if not ugp_player.stream:
+                raise ActionUnavailable(f"UGP player {media.source_id} has no active stream")
             ugp_stream = ugp_player.stream
             # Filter is later applied in MultiClientStream
             audio_source = ugp_stream.get_stream(master_audio_format, filter_params=None)
         elif media.source_id and media.queue_item_id:
             # regular queue stream request
+            queue = self.mass.player_queues.get(media.source_id)
+            queue_item = self.mass.player_queues.get_item(media.source_id, media.queue_item_id)
+            if not queue or not queue_item:
+                raise QueueEmpty("Queue or queue item not found")
             audio_source = self.mass.streams.get_queue_flow_stream(
-                queue=self.mass.player_queues.get(media.source_id),
-                start_queue_item=self.mass.player_queues.get_item(
-                    media.source_id,
-                    media.queue_item_id,
-                ),
+                queue=queue,
+                start_queue_item=queue_item,
                 pcm_format=master_audio_format,
             )
         else:
             # assume url or some other direct path
             # NOTE: this will fail if its an uri not playable by ffmpeg
             audio_source = get_ffmpeg_stream(
                 audio_input=media.uri,
-                input_format=AudioFormat(ContentType.try_parse(media.uri)),
+                input_format=AudioFormat(
+                    content_type=ContentType.try_parse(media.uri) or ContentType.UNKNOWN
+                ),
                 output_format=master_audio_format,
```

Comment:
this whole block is outdated. rebase the branch to find out why

Reply:
OK hang on I see Maxim did something...

Reply:
can you rebase again ?

Reply:
OK I think this is good now...